ios/FluentUI/Vnext/PersonaButtonCarousel/PersonaButtonCarousel.swift

```diff
@@ -5,9 +5,79 @@
 
 import SwiftUI
 
+/// `MSFPersonaCarouselState` contains PersonaCarousel properties
+///
+/// - `buttonSize`: specifies whether to use small or large avatars
+/// - `onTapAction`: provides tap gesture
+@objc public protocol MSFPersonaCarouselState {
+    var buttonSize: MSFPersonaButtonSize { get }
+    var onTapAction: ((_ personaButtonState: MSFPersonaButtonState, _ index: Int) -> Void)? { get set }
+}
+
+/// Properties that make up PersonaGrid content
+class MSFPersonaCarouselStateImpl: NSObject, ObservableObject, Identifiable, MSFPersonaCarouselState {
```

Comment:
If you implement the ID property of the Identifiable protocol assigning a GUID to it, does the animation get smoother?
I think it's worth a shot so that SwiftUI keeps track of the state objects and takes care of the transitions for us.

Reply:
It does not :( I'm not sure why the animation is so janky right now. I'll keep poking at it and see if I can figure anything out.